src/general/DictionaryUtils.cs

```diff
@@ -75,4 +75,49 @@ public static void DivideBy<T>(this Dictionary<T, float> dictionary, float divis
             dictionary[key] /= divisor;
         }
     }
+
+    public static T[] GetKeyArray<T>(this Dictionary<T, long> dictionary)
+    {
+        var keyArray = new T[dictionary.Count];
+        dictionary.Keys.CopyTo(keyArray, 0);
+        return keyArray;
+    }
+
+    public static T[] GetSortedKeyArray<T>(this Dictionary<T, long> dictionary)
```

Comment:
Why do we have manually written sort? What's wrong with LINQ?

Reply:
Nothing but the fact that I just didn't think of it. I just swapped to it